libraries/ESP8266WebServer/src/ESP8266WebServer-impl.h

```diff
@@ -0,0 +1,755 @@
+}
+
+template <class ServerClass, class ClientClass>
+void ESP8266WebServerTemplate<ServerClass, ClientClass>::_handleRequest() {
+  bool handled = false;
+  if (!_currentHandler){
+#ifdef DEBUG_ESP_HTTP_SERVER
+    DEBUG_OUTPUT.println("request handler not found");
+#endif
+  }
+  else {
+    handled = _currentHandler->handle(*this, _currentMethod, _currentUri);
+#ifdef DEBUG_ESP_HTTP_SERVER
+    if (!handled) {
+      DEBUG_OUTPUT.println("request handler failed to handle request");
+    }
+#endif
+  }
+  if (!handled && _notFoundHandler) {
+    _notFoundHandler();
+    handled = true;
+  }
+  if (!handled) {
+    using namespace mime;
+    send(404, String(FPSTR(mimeTable[html].mimeType)), String(("Not found: ")) + _currentUri);
+    handled = true;
+  }
+  if (handled) {
+    _finalizeResponse();
+  }
+  _currentUri = "";
+}
+
+
+template <class ServerClass, class ClientClass>
+void ESP8266WebServerTemplate<ServerClass, ClientClass>::_finalizeResponse() {
+  if (_chunked) {
+    sendContent("");
+  }
+}
+
+template <class ServerClass, class ClientClass>
+String ESP8266WebServerTemplate<ServerClass, ClientClass>::_responseCodeToString(int code) {
+  switch (code) {
+    case 100: return ("Continue");
+    case 101: return ("Switching Protocols");
+    case 200: return ("OK");
+    case 201: return ("Created");
+    case 202: return ("Accepted");
+    case 203: return ("Non-Authoritative Information");
+    case 204: return ("No Content");
+    case 205: return ("Reset Content");
+    case 206: return ("Partial Content");
+    case 300: return ("Multiple Choices");
+    case 301: return ("Moved Permanently");
+    case 302: return ("Found");
+    case 303: return ("See Other");
+    case 304: return ("Not Modified");
+    case 305: return ("Use Proxy");
+    case 307: return ("Temporary Redirect");
+    case 400: return ("Bad Request");
+    case 401: return ("Unauthorized");
+    case 402: return ("Payment Required");
+    case 403: return ("Forbidden");
+    case 404: return ("Not Found");
+    case 405: return ("Method Not Allowed");
+    case 406: return ("Not Acceptable");
+    case 407: return ("Proxy Authentication Required");
+    case 408: return ("Request Time-out");
+    case 409: return ("Conflict");
+    case 410: return ("Gone");
+    case 411: return ("Length Required");
+    case 412: return ("Precondition Failed");
+    case 413: return ("Request Entity Too Large");
+    case 414: return ("Request-URI Too Large");
+    case 415: return ("Unsupported Media Type");
+    case 416: return ("Requested range not satisfiable");
+    case 417: return ("Expectation Failed");
+    case 500: return ("Internal Server Error");
+    case 501: return ("Not Implemented");
+    case 502: return ("Bad Gateway");
+    case 503: return ("Service Unavailable");
+    case 504: return ("Gateway Time-out");
+    case 505: return ("HTTP Version not supported");
+    default:  return ("");
+  }
+}
+
+// AXTLS specialization templates
+template <>
```

Comment:
As mentioned below, rather than implement all of these forwarder methods, how about exposing the _server member via a getter that returns a reference? Then the user can call whatever methods are available in the particular _server object that is instantiated inside.

Reply:
Done in next push